pkg/rancher-desktop/main/deploymentProfiles.ts

```diff
@@ -192,67 +163,256 @@ function parseJsonFiles(rootPath: string, defaultsPath: string, lockedPath: stri
+        nativeReg.closeKey(defaultsKey);
+        nativeReg.closeKey(lockedKey);
+      }
+      // If we found something in the HKLM Defaults or Locked registry hive, don't look at the user's
+      // Alternatively, if the keys work, we could break, even if both hives are empty.
+      if (Object.keys(defaults).length || Object.keys(locked).length) {
+        break;
+      }
+    }
+
+    // Don't bother with the validator, because the registry-based reader validates as it reads.
+    return { defaults, locked };
+  }
+
+  protected fullRegistryPath(...pathParts: string[]): string {
+    return `${ this.keyName }\\${ this.registryPathProfile.join('\\') }\\${ pathParts.join('\\') }`;
+  }
+
+  /**
+   * Windows only. Read settings values from registry using schemaObj as a template.
+   * @param schemaObj the object used as a template for navigating registry.
+   * @param regKey the registry key obtained from nativeReg.openKey().
+   * @param pathParts the relative path to the current registry key, starting at 'Defaults' or 'Locked'
+   * @returns null, or the registry data as an object.
+   */
+  protected readRegistryUsingSchema(schemaObj: Record<string, any>, regKey: nativeReg.HKEY, pathParts: string[]): Record<string, any> {
+    const newObject: Record<string, any> = {};
+    const schemaKeys = Object.keys(schemaObj);
+    const commonKeys: Array<{schemaKey: string, registryKey: string}> = [];
+    const unknownKeys: string[] = [];
+    const userDefinedObjectKeys: Array<{schemaKey: string, registryKey: string}> = [];
+    let regValue: any;
+
+    // Drop the initial 'defaults' or 'locked' field
+    const pathPartsWithoutHiveType = pathParts.slice(1);
+
+    for (const originalKey of nativeReg.enumKeyNames(regKey)) {
+      const schemaKey = fixProfileKeyCase(originalKey, schemaKeys);
+      // "fixed case" means mapping existing keys in the registry (which typically supports case-insensitive lookups)
+      // to the actual case in the schema.
+
+      if (schemaKey === null) {
+        unknownKeys.push(originalKey);
+      } else if (haveUserDefinedObject(pathPartsWithoutHiveType.concat(schemaKey), isEquivalentIgnoreCase)) {
+        userDefinedObjectKeys.push({ schemaKey, registryKey: originalKey });
       } else {
-        const multiSzValue = nativeReg.queryValueRaw(regKey, k);
+        commonKeys.push({ schemaKey, registryKey: originalKey });
+      }
+    }
+    if (unknownKeys.length) {
+      unknownKeys.sort(caseInsensitiveComparator.compare);
+      console.error(`Unrecognized keys in registry at ${ this.fullRegistryPath(...pathParts) }: [${ unknownKeys.join(', ') }]`);
+    }
 
-        if (multiSzValue) {
-          // Registry value can be a single-string or even a DWORD and parseMultiString will handle it.
-          const arrayValue = nativeReg.parseMultiString(multiSzValue as nativeReg.Value);
+    // First process the nested keys, then process any values
+    for (const { schemaKey, registryKey } of commonKeys) {
```

Comment:
We actually never need to use `registryKey` here (because `openKey` and error reporting can both just use `schemaKey` instead), but we don't need to change this.

Reply:
I would rather use `registryKey` in the call to `openKey` because if there's an error I would prefer it to be in the original case in the registry rather than in the schema's case.